lib/hooks/userspace/hook-mongodb-core.js

```diff
@@ -66,32 +65,46 @@ function wrapCallback(span, done) {
   return cls.getNamespace().bind(fn);
 }
 
-module.exports = {
-  patch: function(mongo_, agent_, version_) {
-    if (!semver.satisfies(version_, SUPPORTED_VERSIONS)) {
-      agent_.logger.info('Mongo: unsupported version ' + version_ + ' loaded');
-      return;
-    }
-    if (!mongo) {
-      agent = agent_;
-      mongo = mongo_;
-      shimmer.wrap(mongo.Server.prototype, 'command', wrapWithLabel('mongo-command'));
-      shimmer.wrap(mongo.Server.prototype, 'insert', wrapWithLabel('mongo-insert'));
-      shimmer.wrap(mongo.Server.prototype, 'update', wrapWithLabel('mongo-update'));
-      shimmer.wrap(mongo.Server.prototype, 'remove', wrapWithLabel('mongo-remove'));
-      shimmer.wrap(mongo.Cursor.prototype, 'next', nextWrap);
-    }
-  },
-  unpatch: function() {
-    if (mongo) {
-      shimmer.unwrap(mongo.Server.prototype, 'command');
-      shimmer.unwrap(mongo.Server.prototype, 'insert');
-      shimmer.unwrap(mongo.Server.prototype, 'update');
-      shimmer.unwrap(mongo.Server.prototype, 'remove');
-      shimmer.unwrap(mongo.Cursor.prototype, 'next');
-      agent.logger.info('Mongo: unpatched');
-      mongo = null;
-      agent = null;
-    }
+function onceWrap(once) {
+  return function once_trace(event, cb) {
+    return once.call(this, event, cls.getNamespace().bind(cb));
+  };
+}
+
+module.exports = function(version_, agent_) {
+  if (!semver.satisfies(version_, SUPPORTED_VERSIONS)) {
+    agent_.logger.info('Mongo: unsupported version ' + version_ + ' loaded');
+    return {};
   }
+  return {
+    'lib/connection/pool.js': {
+      patch: function(pool) {
+        shimmer.wrap(pool.prototype, 'once', onceWrap);
```

Comment:
Is there a reason we don't need to wrap `.on`, or alternatively use `bindEmitter` on pool instances?